guillotina/tests/fixtures.py

```diff
@@ -359,7 +359,9 @@ def __init__(self, request):
 async def _clear_dbs(app):
     root = app.root
     # make sure to completely clear db before carrying on...
-    async for db in iter_databases(root):
+    for _, db in root:
```

Comment:
@vangheem I had to change this because it was loading (and creating) the database defined in `settings['storages']['db']` during the cleanup:
```
[37]   /home/jordi/gits/guillotina/guillotina/asgi.py(129)__call__()
-> await self.shutdown()
[38]   /home/jordi/gits/guillotina/guillotina/asgi.py(142)shutdown()
-> await clean(self)
[39]   /home/jordi/gits/guillotina/guillotina/tests/fixtures.py(366)_clear_dbs()
-> if IPostgresStorage.providedBy(storage) or ICockroachStorage.providedBy(storage):
[40]   /home/jordi/gits/guillotina/guillotina/utils/content.py(238)iter_databases()
-> db = await factory.get_database(db_name)
[41]   /home/jordi/gits/guillotina/guillotina/db/factory.py(186)get_database()
-> self.app[name] = await apply_coroutine(factory, name, config)
[42]   /home/jordi/gits/guillotina/guillotina/utils/misc.py(120)apply_coroutine()
-> return await result
[43]   /home/jordi/gits/guillotina/guillotina/db/factory.py(84)PGDatabaseConfigurationFactory()
-> return await _PGConfigurationFactory(key, dbconfig, loop=loop)
[44]   /home/jordi/gits/guillotina/guillotina/db/factory.py(67)_PGConfigurationFactory()
-> aps = storage_factory(**dbconfig)
[45] > /home/jordi/gits/guillotina/guillotina/db/storages/pg.py(524)__init__()
-> self._dsn = dsn
(Pdb++) ll
 515         def __init__(self, dsn=None, partition=None, read_only=False, name=None,                   
 516                      pool_size=13, transaction_strategy='resolve_readcommitted',                   
 517                      conn_acquire_timeout=20, cache_strategy='dummy', db_schema='public',          
 518                      objects_table_name='objects', blobs_table_name='blobs',                       
 519                      connection_manager=None, autovacuum=True, **options):                         
 520             super(PostgresqlStorage, self).__init__(                                               
 521                 read_only, transaction_strategy=transaction_strategy,                              
 522                 cache_strategy=cache_strategy)                                                     
 523             import pdb; pdb.set_trace()                                                            
 524  ->         self._dsn = dsn                                                                        
 525             self._pool_size = pool_size                                                            
 526             self._partition_class = partition                                                      
 527             self._read_only = read_only                                                            
 528             self.__name__ = name                                                                   
 529             self._conn_acquire_timeout = conn_acquire_timeout                                      
 530             self._options = options                                                                
 531             self._connection_options = {}                                                          
 532             self._connection_initialized_on = time.time()                                          
 533             self._db_schema = db_schema                                                            
 534             self._objects_table_name = f'{db_schema}.{objects_table_name}'                         
 535             self._blobs_table_name = f'{db_schema}.{blobs_table_name}'                             
 536             self._sql = SQLStatements()                                                            
 537             self._connection_manager = connection_manager                                          
 538             self._autovacuum = autovacuum                                                          
```
After that, the `DELETE FROM` was crashing with `UndefinedTableError` for this table.

Any idea?

Reply:
Yes, you have it correct. We should not be doing iter_databases there! Good catch.

As for the `DELETE FROM`. Which one? How do I reproduce?

Reply:
You can reproduce it running this test (on commit 5269254c858796e930edec600655bd9023e2aba7):

`DB_SCHEMA=hola DATABASE=postgres pytest -s --cov=guillotina -s --tb=native -v --cov-report term-missing --cov-append guillotina/tests/test_annotations.py -x -s `

Reply:
I'm sorry, maybe I think I misunderstood. I can not reproduce the failure.

I thought you were saying the `DELETE FROM` was still a problem!

Thank you!

Reply:
My bad, removing iter_databases solved the error.

Thank you!